### PR TITLE
chore(jwt-auth): get JWT by ctx.var.cookie_jwt instead of resty.cookie

### PR DESCRIPTION
**Pull request by tzssangglass**
…ookie

Signed-off-by: tzssangglass <tzssangglass@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by tokers** [comment]
@tzssangglass Could you describe the reason for this or add the corresponding issue link?

**Comment by tzssangglass** [comment]
> @tzssangglass Could you describe the reason for this or add the corresponding issue link?

ok, inspired by review this PR is the comments seen: https://github.com/apache/apisix/pull/5727#discussion_r774275326

File: apisix/plugins/jwt-auth.lua [hunk 1]
```diff
@@ -344,8 +341,8 @@ end
 function _M.rewrite(conf, ctx)
     local jwt_token, err = fetch_jwt_token(ctx)
     if not jwt_token then
-        if err and err:sub(1, #"no cookie") ~= "no cookie" then
-            core.log.error("failed to fetch JWT token: ", err)
+        if err then
```
Comment (tokers) [inline apisix/plugins/jwt-auth.lua:344]: Do we still need this check?

Comment (leslie-tsang) [reply apisix/plugins/jwt-auth.lua:344]: Seems we don't need to check `err` as func `fetch_jwt_token` doesn't return it any more.

Comment (tzssangglass) [reply apisix/plugins/jwt-auth.lua:344]: `fetch_jwt_token` would return `"JWT not found in cookie"`

Comment (tzssangglass) [reply apisix/plugins/jwt-auth.lua]: update

**Review by leslie-tsang — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Review by shuaijinchao — APPROVED** [review]

**Pull request merged**